foundry/pixi/containers/ruler.d.ts

```diff
@@ -1,167 +1,184 @@
+   * @param destination - The destination point to which to measure
+   * @param gridSpaces  - Restrict measurement only to grid spaces
+   *                      (Default value: `true`)
    */
-  public waypoints: PIXI.Point[];
+  measure(
+    destination: Point,
+    { gridSpaces }?: { gridSpaces?: boolean }
+  ): {
+    ray: Ray;
+    label: PIXI.DisplayObject;
+  }[];
 
-  constructor(user: User, color?: number);
+  /**
+   * Get the text label for a segment of the measured path
+   */
+  protected _getSegmentLabel(segmentDistance: number, totalDistance: number, isTotal?: boolean): string;
 
   /**
-   * Acquire a Token, if any, which is eligible to perform a movement based on the starting point of the Ruler
-   * @internal
+   * Highlight the measurement required to complete the move in the minimum number of discrete spaces
    */
-  public _getMovementToken(): Token;
+  protected _highlightMeasurement(ray: Ray): void;
 
   /**
-   * A helper method to return an Array of Ray objects constructed from the waypoints of the measurement
-   * @param waypoints - An Array of waypoint \{x, y\} Objects
-   * @param destination - An optional destination point to append to the existing waypoints
-   * @returns An Array of Ray objects which represent the segemnts of the waypoint path
-   * @internal
+   * Determine whether a SPACE keypress event entails a legal token movement along a measured ruler
+   *
+   * @returns An indicator for whether a token was successfully moved or not. If True the event should be
+   *          prevented from propagating further, if False it should move on to other handlers.
+   * @remarks This will never actually return `true`
    */
-  public _getRaysFromWaypoints(waypoints: PIXI.Point[], destination: PIXI.Point): Ray[];
+  moveToken(): Promise<boolean | undefined>;
 
   /**
-   * Get the text label for a segment of the measured path
-   * @param ray -
-   * @param segmentDistance -
-   * @param totalDistance -
-   * @param isTotal -
-   * @internal
+   * Acquire a Token, if any, which is eligible to perform a movement based on the starting point of the Ruler
    */
-  public _getSegmentLabel(ray: Ray, segmentDistance: number, totalDistance: number, isTotal: boolean): string;
+  protected _getMovementToken(): Token | null | undefined;
 
   /**
-   * Highlight the measurement required to complete the move in the minimum number of discrete spaces
-   * @param ray -
-   * @internal
+   * A helper method to return an Array of Ray objects constructed from the waypoints of the measurement
+   * @param waypoints   - An Array of waypoint `{x, y}` Objects
+   * @param destination - An optional destination point to append to the existing waypoints
+   * @returns             An Array of Ray objects which represent the segemnts of the waypoint path
    */
-  public _highlightMeasurement(ray: Ray): void;
+  protected _getRaysFromWaypoints(waypoints: { x: number; y: number }[], destination?: { x: number; y: number }): Ray[];
```

Comment:
Maybe use `Point[]` and `Point` as the types for the parameters?

Reply:
`Point` is a union type that includes `PIXI.Point`.